File: frontend/src/types/handoff.ts
export type HandoffStatus = 'PENDING' | 'IN_PROGRESS' | 'COMPLETED';

export interface Handoff {
  id: number;
  emergency_case_id: number;
  hospital_id: number;
  ambulance_id: number;
  arrival_time?: string;
  handoff_time?: string;
  received_by?: string;
  notes?: string;
  status: HandoffStatus;
  created_at: string;
  updated_at: string;
}

export interface HandoffCompleteRequest {
  received_by: string;
  notes?: string;
}

File: backend/app/schemas/handoff.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict
from app.models.handoff import HandoffStatus

class HandoffBase(BaseModel):
    emergency_case_id: int
    hospital_id: int
    ambulance_id: int

class HandoffRead(HandoffBase):
    id: int
    arrival_time: Optional[datetime] = None
    handoff_time: Optional[datetime] = None
    received_by: Optional[str] = None
    notes: Optional[str] = None
    status: HandoffStatus
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

class HandoffCompleteRequest(BaseModel):
    received_by: str
    notes: Optional[str] = None
